storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java

```diff
@@ -0,0 +1,355 @@
+            } else {
+                stat = zk.checkExists().forPath(Utils.normalizePath(path));
+            }
+            return Integer.valueOf(stat.getVersion());
+        }
+
+        return null;
+    }
+
+    public static List<String> getChildren(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            if (watch) {
+                return zk.getChildren().watched().forPath(npath);
+            } else {
+                return zk.getChildren().forPath(npath);
+            }
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    // Deletes the state inside the zookeeper for a key, for which the
+    // contents of the key starts with nimbus host port information
+    public static void deleteDodeBlobstore(CuratorFramework zk, String parentPath, String hostPortInfo) throws RuntimeException {
+        String parentnPath = Utils.normalizePath(parentPath);
+        List<String> childPathList = null;
+        if (existsNode(zk, parentnPath, false)) {
+            childPathList = getChildren(zk, parentnPath, false);
+        }
+        for (String child : childPathList) {
+            if (child.startsWith(hostPortInfo)) {
+                LOG.debug("deleteNode child " + child);
+                deleteNode(zk, parentnPath + "/" + child);
+            }
+        }
+    }
+
+    public static Stat setData(CuratorFramework zk, String path, byte[] data) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            return zk.setData().forPath(npath, data);
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    public static boolean exists(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+        return existsNode(zk, path, watch);
+    }
+
+    public static NIOServerCnxnFactory mkInprocessZookeeper(String localdir, Integer port) throws IOException, InterruptedException {
+        LOG.info("Starting inprocess zookeeper at port " + port + " and dir " + localdir);
+        File localfile = new File(localdir);
+        ZooKeeperServer zk = new ZooKeeperServer(localfile, localfile, 2000);
+
+        NIOServerCnxnFactory factory = null;
+        int report = 2000;
+        int limitPort = 65535;
+        if (port != null) {
+            report = port;
+            limitPort = port;
+        }
+        while (true) {
+            try {
+                factory = new NIOServerCnxnFactory();
+                factory.configure(new InetSocketAddress(port), 0);
+                break;
+            } catch (BindException e) {
+                report++;
+                if (report > limitPort) {
+                    throw new RuntimeException("No port is available to launch an inprocess zookeeper");
+                }
+            }
+        }
+        factory.startup(zk);
+        return factory;
```

Comment:
Are you sure that we don't need the retport?  I see that factory has a getLocalPort method is that good enough?